daemon/cmd/fqdn.go

```diff
@@ -393,3 +313,124 @@ func (d *Daemon) notifyOnDNSMsg(
 
 	return nil
 }
+
+func (d *Daemon) updateOnDNSMsg(lookupTime time.Time, ep *endpoint.Endpoint, qname string, responseIPs []netip.Addr, TTL int, stat *dnsproxy.ProxyRequestContext) error {
+	if stat != nil {
```

Comment:
why would stat be nil? This check wasn't in the code prior to extracting it

Reply:
Since both embedded dns proxy and standalone dns proxy are using this method. And standalone dns proxy does not send the `stats` as of now because of the current grpc contract. I added the `nil` check.